### ISSUE TITLE
Excess query executed against SQLite when emulating INSERT .. RETURNING

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14328]
The SQLite emulation of `INSERT .. RETURNING` always runs 2 queries:

```sql
SELECT last_insert_rowid();
SELECT * FROM t WHERE _rowid_ = :last_insert_rowid;
```

This is caused by a query in `AbstractDMLQuery`:

```java
create.select(returning)
      .from(table)
      .where(rowid().equal(rowid().getDataType().convert(create.lastID())))
      .fetch();
```

The `create.lastID()` call is a roundtrip to the database, not an expression. See also: https://github.com/jOOQ/jOOQ/issues/12823

**Comment by lukaseder** [comment]
A 3rd query used to be executed on `UpdatableRecord::store` calls, when `withReturnAllOnUpdatableRecord(true)`! See https://github.com/jOOQ/jOOQ/issues/11620

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14329), 3.16.12 (https://github.com/jOOQ/jOOQ/issues/14330)
